pkg/network/protocols/testutil/serverutils.go

```diff
@@ -81,11 +81,12 @@ func runDockerServer(t testing.TB, serverName, dockerPath string, env []string,
 	t.Cleanup(func() {
 		cancel()
 		_ = cmd.Wait()
+		timedContext, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
nit: use `defer` for the cancellation function instead
```suggestion
		timedContext, cancel := context.WithTimeout(context.Background(), timeout)
		defer cancel()
```